florestad/src/json_rpc/blockchain.rs

```diff
@@ -188,13 +191,64 @@ impl<Blockchain: RpcChain> RpcImpl<Blockchain> {
         res.map_err(|_e| RpcError::Encode)
     }
 
-    // gettxoutproof
-    pub(super) fn get_tx_proof(&self, tx_id: Txid) -> Result<Vec<String>, RpcError> {
-        Ok(self
-            .wallet
-            .get_merkle_proof(&tx_id)
-            .ok_or(RpcError::TxNotFound)?
-            .0)
+    /// Return the block that contains the given Txid
+    pub fn get_block_by_txid(&self, txid: &Txid) -> Result<Block, RpcError> {
+        let height = self.wallet.get_height(txid).ok_or(RpcError::TxNotFound)?;
+        let blockhash = self.chain.get_block_hash(height).unwrap();
+        Ok(self.chain.get_block(&blockhash).unwrap())
+    }
```

Comment:
The way I was organizing this, all helper methods should be in the first `impl` block. In the second block we should only have the RPC methods.